docs/vscode.md

```diff
@@ -464,6 +464,34 @@ process greet {
 }
 ```
 
+**Workflow onComplete/onError handlers**
+
+{ref}`Workflow handlers <workflow-handlers>` (i.e. `workflow.onComplete` and `workflow.onError`) can be defined in several different ways in a script, but are typically defined as top-level statements and without an equals sign:
+
+```nextflow
+workflow.onComplete {
+    println "Pipeline completed at: $workflow.complete"
+    println "Execution status: ${ workflow.success ? 'OK' : 'failed' }"
+}
+```
+
+The Nextflow language specification does not allow statements to be mixed with script declarations, so workflow handlers must be defined in the entry workflow:
+
+```nextflow
+workflow {
+    // ...
+
+    workflow.onComplete = {
+        println "Pipeline completed at: $workflow.complete"
+        println "Execution status: ${ workflow.success ? 'OK' : 'failed' }"
+    }
```

Comment:
Bit ugly nested `workflow`. In principle `onComplete` could be considered a `workflow` attribute, as such, be declarable outside the main `workflow` (and even desirable for better separation of concerns)

Reply:
This is a temporary fix until we provide a better syntax. People keep asking me about it so it needs to be documented

Reply:
Fair enough